includes/class-wc-post-data.php

```diff
@@ -254,7 +254,10 @@ public static function wp_insert_post_data( $data ) {
 					$data['post_parent'] = 0;
 				break;
 			}
+		} elseif ( 'product' === $data['post_type'] && 'auto-draft' === $data['post_status'] ) {
```

Comment:
Are there cases where this won't run, since the previous if/else is:

`} elseif ( 'product' === $data['post_type'] && isset( $_POST['product-type'] ) ) {`